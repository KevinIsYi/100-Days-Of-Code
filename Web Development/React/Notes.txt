{Go to folder}
npx create-react-app `name`
cd `name`
npm start

rafcp 
import React from 'react'
import PropTypes from 'prop-types'

const firstApp = props => {
    return (
        <div>
            
        </div>
    )
}

firstApp.propTypes = {

}

export default firstApp