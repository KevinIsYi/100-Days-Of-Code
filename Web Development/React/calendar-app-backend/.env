PORT=4000
DB_CONNECTION=mongodb+srv://mern_user:NekxHA9lixWzsT0c@cluster0.9xvmk.mongodb.net/mern_calendar
JWT_SEED=p@L@BR@_S3CR3T@_JWT