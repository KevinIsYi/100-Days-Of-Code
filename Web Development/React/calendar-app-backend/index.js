const express = require('express');
const dotenv = require('dotenv');
const cors = require('cors');
const { dbConnection } = require('./database/config');

dotenv.config();

// Create express server
const app = express();

// Database
dbConnection();


//CORS
app.use(cors());

app.use(express.static('public'));

// Lecture and parse of body
app.use( express.json() );

app.use('/api/auth', require('./routes/auth'));
app.use('/api/events', require('./routes/events'));


// Listen petitions
app.listen(process.env.PORT, () => { // Port, callback
    console.log(`Server is running on: ${ process.env.PORT }`);
});