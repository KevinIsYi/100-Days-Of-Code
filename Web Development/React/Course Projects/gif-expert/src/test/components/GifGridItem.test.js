import React from 'react';
const { shallow } = require("enzyme");
const { GifGridItem } = require("../../components/GifGridItem");

describe("Test on <GifGridItem>", () => {

    const url = "https//gdsgdsdjkgjkds.com"
    const wrapper = shallow(<GifGridItem url={url}/>);
    test("Must match snapshot", () => {
        

        expect(wrapper).toMatchSnapshot();
    });

    test("url must match with what is send", () => {
        const img = wrapper.find("img");

        expect(img.prop("src")).toBe(url);
    });

    test("classes in div must be animate__fadeInLeft", () => {
        const img = wrapper.find("img");

        //console.log(img.props());
        expect(img.prop("className")).toBe("animate__fadeInLeft");
    });
});