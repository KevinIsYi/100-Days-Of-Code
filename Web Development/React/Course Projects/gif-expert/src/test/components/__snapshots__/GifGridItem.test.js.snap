// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGridItem> Must match snapshot 1`] = `
<img
  alt="gif"
  className="animate__fadeInLeft"
  src="https//gdsgdsdjkgjkds.com"
/>
`;
